system/lib/pthread/library_pthread_stub.c

```diff
@@ -139,14 +139,16 @@ int pthread_setspecific(pthread_key_t key, const void* value) {
 /*magic number to detect if we have not run yet*/
 #define PTHREAD_ONCE_MAGIC_ID 0x13579BDF
 
-int pthread_once(pthread_once_t* once_control, void (*init_routine)(void)) {
+int __pthread_once(pthread_once_t* once_control, void (*init_routine)(void)) {
   if (*once_control != PTHREAD_ONCE_MAGIC_ID) {
     init_routine();
     *once_control = PTHREAD_ONCE_MAGIC_ID;
   }
   return 0;
 }
 
+weak_alias(__pthread_once, pthread_once);
```

Comment:
what's going on with `pthread_once` here?

Reply:
If you include libc with `--whole-archive` it ends up calling `__pthread_once` (from system/lib/libc/musl/src/thread/call_once.c).   The only time we were doing `--whole-archive` in practice was with `MAIN_MODULE=1` and we were not reported errors on undefined symbols before. (the LINKABLE setting was causing all error to be suppressed, now it isn't).